File: app/src/main/java/kr/djspi/pipe01/BaseApplication.java
package kr.djspi.pipe01;

import android.app.Application;

import com.squareup.leakcanary.LeakCanary;

public class BaseApplication extends Application {

    private static final boolean USE_LEAK_CANARY = true;

    @Override
    public void onCreate() {
        super.onCreate();
        if (USE_LEAK_CANARY) {
            if (LeakCanary.isInAnalyzerProcess(this)) {
                // This process is dedicated to LeakCanary for heap analysis.
                // You should not init your app in this process.
                return;
            }
            LeakCanary.install(this);
            // Normal app init code...
        }
    }
}

File: app/src/main/java/kr/djspi/pipe01/BaseApplication.kt
package kr.djspi.pipe01

import android.app.Application
import com.squareup.leakcanary.LeakCanary

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        if (USE_LEAK_CANARY) {
            if (LeakCanary.isInAnalyzerProcess(this)) return
            LeakCanary.install(this)
        }
    }

    companion object {
        private const val USE_LEAK_CANARY = false
    }
}